src/core/src/op/topk.cpp

```diff
@@ -169,156 +160,84 @@ void op::v1::TopK::k_type_check(const element::Type& k_element_type) const {
         ").");
 }
 
-shared_ptr<Node> op::v1::TopK::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> TopK::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_TopK_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<v1::TopK>(new_args.at(0), new_args.at(1), m_axis, m_mode, m_sort, m_index_element_type);
+    return std::make_shared<TopK>(new_args.at(0), new_args.at(1), m_axis, m_mode, m_sort, m_index_element_type);
 }
 
-bool op::v1::TopK::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool TopK::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_TopK_evaluate);
-    return topk::TopK_evaluate(this, outputs, inputs);
+    return topk::evaluate(this, outputs, inputs);
 }
 
-bool op::v1::TopK::has_evaluate() const {
+bool TopK::has_evaluate() const {
     OV_OP_SCOPE(v1_TopK_has_evaluate);
-
-    switch (get_input_element_type(0)) {
-    case element::i32:
-    case element::i64:
-    case element::u32:
-    case element::u64:
-    case element::f16:
-    case element::f32:
-        break;
-    default:
-        return false;
-    }
-
-    if (op::util::is_constant(input_value(1).get_node())) {
```

Comment:
@jane-intel @ilyachur 
I've unified the type check for input1. 
Do you know why it was different for constant and not constant input?

Reply:
I don't remember @jane-intel Could you comment?